tests/core/MantisTestCase.php

```diff
@@ -64,4 +64,35 @@ function getTestName(): string {
 
 		return $t_class_name . '::' . $t_test_method;
 	}
+
+	/**
+	 * Sets the given configuration and returns its old value.
+	 *
+	 * @param string $p_config Configuration option name.
+	 * @param mixed  $p_value  Configuration option value.
+	 *
+	 * @return mixed The config's old value, false if it was not set in the database.
+	 */
+	public function setConfig( string $p_config, $p_value ) {
+		$t_old = config_is_set_in_database( $p_config ) ? config_get( $p_config ) : false;
+		config_set( $p_config, $p_value );
+
+		return $t_old;
+	}
+
+	/**
+	 * Restores a configuration to its initial state.
+	 *
+	 * @param string $p_config Configuration option name.
+	 * @param mixed  $p_value  Configuration option value. If false, the config
+	 *                         will be deleted.
+	 */
+	public function restoreConfig( string $p_config, $p_value ) {
+		if( $p_value === false ) {
```

Comment:
Isn't it the case that some configs may be enabled by default? Hence, false doesn't mean deletion?

Reply:
Technically it is possible if a config value is defined as boolean false. The logic relies on our use ON/OFF constants in the config files, corresponding to *integer* values 0/1.

IMO considering the limited use cases for this, it was not worth implementing something more complex for a fully general usage.